libsolidity/parsing/Parser.cpp

```diff
@@ -102,6 +102,9 @@ ASTPointer<SourceUnit> Parser::parse(shared_ptr<Scanner> const& _scanner)
 			case Token::Enum:
 				nodes.push_back(parseEnumDefinition());
 				break;
+			case Token::Function:
+				nodes.push_back(parseFunctionDefinition());
```

Comment:
This will currently allow functions without body. Not sure if this is intentional. How will they get implemented?

Reply:
This PR was just a test if it can work - we need to add more restrictions.